control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -1280,65 +1286,51 @@ func (r *HostedControlPlaneReconciler) reconcileCloudProviderConfig(ctx context.
 func (r *HostedControlPlaneReconciler) reconcileManagedEtcd(ctx context.Context, hcp *hyperv1.HostedControlPlane, releaseImage *releaseinfo.ReleaseImage) error {
 	p := etcd.NewEtcdParams(hcp, releaseImage.ComponentImages())
 
-	// Etcd Operator ServiceAccount
-	operatorServiceAccount := manifests.EtcdOperatorServiceAccount(hcp.Namespace)
-	if _, err := r.CreateOrUpdate(ctx, r, operatorServiceAccount, func() error {
-		return etcd.ReconcileOperatorServiceAccount(operatorServiceAccount, p.OwnerRef)
+	discoveryService := manifests.EtcdDiscoveryService(hcp.Namespace)
+	if result, err := controllerutil.CreateOrUpdate(ctx, r, discoveryService, func() error {
+		return etcd.ReconcileDiscoveryService(discoveryService, p.OwnerRef)
 	}); err != nil {
-		return fmt.Errorf("failed to reconcile etcd operator service account: %w", err)
+		return fmt.Errorf("failed to reconcile etcd discovery service: %w", err)
+	} else {
+		r.Log.Info("reconciled etcd discovery service", "result", result)
```

Comment:
This will end up printing a log message whenever the reconciler got an event on any of the things it watches, do we really want that? Same for the logs below